libs/blocks/quiz/utils.js

```diff
@@ -0,0 +1,441 @@
+    );
+  });
+
+  nestedFragsSecondaryArray?.forEach((frag) => {
+    if (!frag) return;
+    const fragKey = frag.trim();
+    if (!nestedObject[fragKey]) nestedObject[fragKey] = [];
+    nestedObject[fragKey].push(...getNestedFragments(
+      resultResources,
+      secondaryProductCodes,
+      fragKey,
+    ));
+  });
+
+  return nestedObject;
+};
+
+const getNestedFragments = (resultResources, productCodes, fragKey) => {
+  const fragArray = [];
+  productCodes?.forEach((product) => {
+    resultResources?.data?.forEach((row) => {
+      if (product && product === row?.product) {
+        insertFragment();
+      }
+
+      function insertFragment() {
+        if (row[fragKey]) {
+          row[fragKey]?.split(',').forEach((val) => {
+            fragArray.push(val.trim());
+          });
+        }
+      }
+    });
+  });
+  return fragArray;
+};
+
+export const getRedirectUrl = (destinationPage, primaryProducts) => {
+  const separator = destinationPage.includes('?') ? '&' : '?';
+  return `${destinationPage}${separator}primary=${primaryProducts}&quizKey=${quizKey}`;
+};
+
+export const parseResultData = async (answers) => {
+  const results = await fetchContentOfFile(RESULTS_EP_NAME);
+  const filteredResults = results.result.data.reduce(
+    (resultObj, resultMap) => {
+      let hasMatch = false;
+      const resultRow = Object.entries(resultMap);
+
+      for (let i = 0; i < resultRow.length; i += 1) {
+        const key = resultRow[i][0];
+        const val = resultRow[i][1];
+
+        if (!key.startsWith('result-') && val) {
+          const answer = answers.find((a) => a[0] === key);
+          if (answer && answer[1].includes(val)) {
+            hasMatch = true;
+          } else {
+            // q-question value did not match any answers, entire row does not match
+            return resultObj;
+          }
+        }
+      }
+
+      if (hasMatch) {
+        resultObj.primary = resultObj.primary.concat(
+          resultMap['result-primary'].split(','),
+        );
+        resultObj.secondary = resultObj.secondary.concat(
+          resultMap['result-secondary'].split(','),
+        );
+      }
+
+      return resultObj;
+    },
+    {
+      primary: [],
+      secondary: [],
+    },
+  );
+
+  filteredResults.primary = [...new Set(filteredResults.primary.filter(Boolean))];
+  filteredResults.secondary = [...new Set(filteredResults.secondary.filter(Boolean))];
+
+  // Find result destination page.
+  const matchedResults = findMatchForSelections(
+    results['result-destination'].data,
+    filteredResults,
+  );
+
+  filteredResults.matchedResults = matchedResults;
+  const rObj = {};
+  rObj.filteredResults = filteredResults;
+  rObj.resultResources = results['result-fragments'];
+
+  return rObj;
+};
+
+const getRecommendedResults = (selectedDestination, defaultValue) => (selectedDestination.length
+  ? selectedDestination : defaultValue);
+
+// TODO: needs refactoring - can split to smaller functions
```

Comment:
Remove comment.  Create an issue to track it if it's important that you refactor it.